charmcraft/application/commands/test.py

```diff
@@ -84,6 +90,9 @@ def run(self, parsed_args: argparse.Namespace):
         else:
             cmd = ["spread"]
 
+        if vars(parsed_args).get("verbose"):
```

Comment:
```suggestion
        if emit.get_mode() >= EmitterMode.VERBOSE:
```

(You'll need to import `EmitterMode` from craft_cli above, but then you can remove the `add_argument` too)

Reply:
Could you also add tests? This unit test should be fine in `tests/unit/commands/test_test.py`:

```python
# Copyright 2024 Canonical Ltd.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# For further info, check https://github.com/canonical/charmcraft
"""Unit tests for the test command."""


import argparse
from craft_cli import EmitterMode, emit
import pytest

from charmcraft.application.commands.test import TestCommand
from charmcraft.application.main import APP_METADATA


@pytest.mark.parametrize("emitter_mode", [EmitterMode.VERBOSE, EmitterMode.DEBUG, EmitterMode.TRACE])
def test_spread_verbose(fp, capsys, service_factory, emitter_mode: EmitterMode) -> None:
    emit._mode = emitter_mode
    fp.register(["spread", "-vv", "wololo"])
    cmd = TestCommand({"app": APP_METADATA, "services": service_factory})
    args = argparse.Namespace(spread_tasks=["wololo"])

    cmd.run(args)


@pytest.mark.parametrize("emitter_mode", [EmitterMode.QUIET, EmitterMode.BRIEF])
def test_spread_not_verbose(fp, capsys, service_factory, emitter_mode: EmitterMode) -> None:
    emit._mode = emitter_mode
    fp.register(["spread", "wololo"])
    cmd = TestCommand({"app": APP_METADATA, "services": service_factory})
    args = argparse.Namespace(spread_tasks=["wololo"])

    cmd.run(args)

```